src/main/java/org/zalando/nakadi/service/subscription/zk/NewZkSubscriptionClient.java

```diff
@@ -216,4 +218,50 @@ public void transfer(final String sessionId, final Collection<EventTypePartition
                     changeSet.toArray(new Partition[changeSet.size()]));
         }
     }
+
+    @Override
+    public void repartitionTopology(final String eventTypeName, final int newPartitionsCount)
+            throws NakadiRuntimeException {
+        final Topology currentTopology = getTopology();
+        final List<Partition> partitionsList = Lists.newArrayList(currentTopology.getPartitions());
+        final List<Partition> newPartitionsList = new ArrayList<>();
+
+        // Set other event type's partition state to unassigned
+        for (final Partition partition : partitionsList) {
```

Comment:
why do we set current partitions in use to unassigned state?

Reply:
If I understood correctly, we should disconnect all the sessions using this subscription, and they reconnect again? (If this is the case, then maybe I should clean the `/subscriptions/<id>/sessions` node too.)

Reply:
This will not help with reconnect, because you also should wait for commit to come until commit timeout. I would not go this way, because it changes internal structure at runtime, I would rather do it the same way it is done for reset cursor, but it is work for another pr.

Reply:
Got it, thanks! :), fixed this in `d0df4e5`.